### PR TITLE
[BS-14134] catch OutOfMemoryError on big file upload

### PR DESCRIPTION
**Pull request by redboul**
to response appropriate message

### CODE REVIEW

File: server/src/main/java/org/bonitasoft/console/common/server/servlet/FileUploadServlet.java [hunk 1]
```diff
@@ -134,7 +135,17 @@ public void doPost(final HttpServletRequest request, final HttpServletResponse r
...
             final ServletFileUpload serviceFileUpload = createServletFileUpload(fileItemFactory);
-            final List<FileItem> items = serviceFileUpload.parseRequest(request);
+            if (checkUploadedFileSize) {
+                setUploadSizeMax(serviceFileUpload, request);
+            }
+            List<FileItem> items;
+            try {
+                items = serviceFileUpload.parseRequest(request);
+            } catch (final OutOfMemoryError e) {
+                throw new SizeLimitExceededException("The file exceeds its maximum permitted size.",
```
Comment (apupier) [inline server/src/main/java/org/bonitasoft/console/common/server/servlet/FileUploadServlet.java]: there is no derefencing of object after the catch of the OOM. The JVM might not be able to have enough meomry to step back to a working state. Maybe try to set serviceFileUpload to null and call a GC would be safer (but would it be enough? Who is handling the big chunk of memory?)

File: server/src/main/java/org/bonitasoft/console/common/server/servlet/TenantFileUploadServlet.java [hunk 2]
```diff
@@ -49,16 +47,8 @@ protected APISession getAPISession(final HttpServletRequest request) {
...
-        final long maxSize = getConsoleProperties(getAPISession(request).getTenantId()).getMaxSize();
-        if (contentSize > maxSize * 1048576) {
-            final String errorMessage = "file " + item.getName() + " too big !";
-            if (LOGGER.isLoggable(Level.SEVERE)) {
-                LOGGER.log(Level.SEVERE, errorMessage);
-            }
-            throw new FileTooBigException(errorMessage, item.getName(), String.valueOf(maxSize));
-        }
+    protected void setUploadSizeMax(final ServletFileUpload serviceFileUpload, final HttpServletRequest request) {
+        serviceFileUpload.setFileSizeMax(getConsoleProperties(getAPISession(request).getTenantId()).getMaxSize() * 1048576);
```
Comment (apupier) [inline server/src/main/java/org/bonitasoft/console/common/server/servlet/TenantFileUploadServlet.java]: 1048576: Where this number come from?
You shoudl use a constant which will explained it

File: server/src/test/java/org/bonitasoft/console/common/server/servlet/TenantFileUploadServletTest.java [hunk 3]
```diff
@@ -1,7 +1,7 @@
 package org.bonitasoft.console.common.server.servlet;
```
Comment (apupier) [inline server/src/test/java/org/bonitasoft/console/common/server/servlet/TenantFileUploadServletTest.java]: missing header

File: server/src/main/java/org/bonitasoft/console/common/server/servlet/FileUploadServlet.java [hunk 4]
```diff
@@ -195,6 +200,23 @@ public void doPost(final HttpServletRequest request, final HttpServletResponse r
         }
     }
 
+    /**
+     * @param response
+     * @param responsePW
+     * @param e
```
Comment (apupier) [inline server/src/main/java/org/bonitasoft/console/common/server/servlet/FileUploadServlet.java]: param e has been renamed to message

File: server/src/main/java/org/bonitasoft/console/common/server/servlet/FileUploadServlet.java (hunk 1 above)
Comment (apupier) [inline server/src/main/java/org/bonitasoft/console/common/server/servlet/FileUploadServlet.java]: I feel that the erro message is not correct. it is not that it exceeds its permitted size but the file exceeds the current server capacity.
The permitted size shoudl be handled by setUploadSizeMax( (if I understood well)

**Comment by abirembaut** [comment]
Same comment as on PR203 : Did you check that there is no side-effect client-side (I think there is some client code parsing the error and displaying a specific message in case of a FileTooBigException (at least in forms and maybe in portal)) ?

**Comment by redboul** [comment]
In fact, when the OOM error happens, no response is returned and tomcat just get rid of the resources, the request stays pending in the browser...

File: server/src/main/java/org/bonitasoft/console/common/server/servlet/TenantFileUploadServlet.java (hunk 2 above)
Comment (redboul) [inline server/src/main/java/org/bonitasoft/console/common/server/servlet/TenantFileUploadServlet.java]: it's 1Mo as octet value

File: server/src/main/java/org/bonitasoft/console/common/server/servlet/FileUploadServlet.java (hunk 1 above)
Comment (redboul) [inline server/src/main/java/org/bonitasoft/console/common/server/servlet/FileUploadServlet.java]: I tried both setter and it the error keeps raising. 
The solution I'm trying to implement is for the servlet to try to alert the user that some error happens. Currently, when this takes place, nothing is thrown back to the browser...

File: server/src/main/java/org/bonitasoft/console/common/server/servlet/TenantFileUploadServlet.java (hunk 2 above)
Comment (apupier) [inline server/src/main/java/org/bonitasoft/console/common/server/servlet/TenantFileUploadServlet.java]: yes I know, it was rhetoric.
We shoudl add a constant and perhaps even use a convertToOctet method

File: server/src/test/java/org/bonitasoft/console/common/server/servlet/TenantFileUploadServletTest.java (hunk 3 above)
Comment (redboul) [inline server/src/test/java/org/bonitasoft/console/common/server/servlet/TenantFileUploadServletTest.java]: done

**Pull request merged**